src/sentry/search/events/fields.py

```diff
@@ -2647,18 +2647,33 @@ def resolve_select(self, selected_columns: Optional[List[str]]) -> List[SelectTy
         if selected_columns is None:
             return []
 
-        columns = []
+        resolved_columns = []
 
-        for column in selected_columns:
-            if column.strip() == "":
+        stripped_columns = [column.strip() for column in selected_columns]
+
+        # Add threshold config alias if there's a function that dependso on it
+        for column in {
+            "apdex()",
+            "count_miserable(user)",
+            "user_misery()",
+        }:
+            if (
+                column in stripped_columns
+                and PROJECT_THRESHOLD_CONFIG_ALIAS not in stripped_columns
+            ):
+                stripped_columns.append(PROJECT_THRESHOLD_CONFIG_ALIAS)
+                break
```

Comment:
We shouldn't need this in SnQL right? It is automatically resolved and aliased (and not a selected column) when apdex or count_miserable or user_misery are resolved. And should never have to query or filter on `PROJECT_THRESHOLD_CONFIG_ALIAS`

Reply:
And if that's the case, then the only change this PR would need is changing the `self.columns` [here](https://github.com/getsentry/sentry/pull/27980/files#diff-e7d5c143e6b668a98f4dc22ec4fdc3796c413d23cfbd4d3d43c1cd001465cddeL2658)

Reply:
Don't we use the threshold in the response on the frontend? or did I misread this code here https://github.com/getsentry/sentry/blob/master/static/app/utils/discover/fieldRenderers.tsx#L472

Reply:
I think you're right that we need the project threshold on the frontend. In that case, can we make it do something like conditionally insert if the project column/transaction column exists? Because if we always insert it, we'll never be able to delete the averaging hack for charts.

Reply:
Ahh I forgot about using it on the frontend. yeah we conditionally add so that we don't add it for the timeseries charts